doc: hpf and softperipherals comparison (#23226)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54l/vpr_flpr.rst

When building for the ``nrf54l15dk/nrf54l15/cpuflpr`` target, a minimal sample is automatically loaded onto the application core.
See more information on :ref:`building_nrf54l_app_flpr_core`.

Using FLPR to emulate peripherals
*********************************

.. caution::

   The High-Performance Framework (HPF) support in the |NCS| is :ref:`experimental <software_maturity>` and is limited to the nRF54L15 device.

The FLPR core can emulate software peripherals using either Soft Peripherals or the High-Performance Framework (HPF).
This setup is useful in scenarios where additional peripheral functionality is needed or when aiming to optimize power consumption and performance.
For detailed comparisons of implementation scenarios and use cases between the Soft Peripherals and the High-Performance Framework (HPF), refer to the  :ref:`ug_hpf_softperipherals_comparison` documentation page.

> Using MCUboot with FLPR

Reviewer: See the comment about gerunds above.

When building for the ``nrf54l15dk/nrf54l15/cpuflpr`` target, a minimal sample is automatically loaded onto the application core.
See more information on :ref:`building_nrf54l_app_flpr_core`.

Peripherals emulation on FLPR
*****************************

> .. caution::

Reviewer 2: Wouldn't a note be enough?

   The High-Performance Framework (HPF) support in the |NCS| is :ref:`experimental <software_maturity>` and is limited to the nRF54L15 device.

> The FLPR core can emulate software peripherals using either Soft Peripherals or the High-Performance Framework (HPF).

Reviewer 2: ```suggestion
The FLPR core can emulate software peripherals using either Soft Peripherals or the HPF.
```

> This setup is useful in scenarios where you need additional peripheral functionality is needed or when aiming to optimize power consumption and performance.

Reviewer 3: ```suggestion
This setup is useful in scenarios where you need additional peripheral functionality or want to optimize power consumption and performance.
```

> For detailed comparisons of implementation scenarios and use cases between the Soft Peripherals and the High-Performance Framework (HPF), refer to the  :ref:`ug_hpf_softperipherals_comparison` documentation page.

Reviewer 2: ```suggestion
For detailed comparisons of implementation scenarios and use cases between the Soft Peripherals and the HPF, refer to the :ref:`ug_hpf_softperipherals_comparison` documentation page.
```
